### PR TITLE
gate the stable_hash test to only run on little endinan machines

### LINKED ISSUE #39: cache_tests::stable_hash test fails on s390x
Output:

```
---- cache_tests::stable_hash stdout ----
thread 'cache_tests::stable_hash' panicked at 'assertion failed: `(left == right)`
  left: `"038D651019410545"`,
 right: `"7D208C81E8236995"`', src/lib.rs:869:9
stack backtrace:
   0: rust_begin_unwind
             at /usr/src/rustc-1.63.0/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /usr/src/rustc-1.63.0/library/core/src/panicking.rs:142:14
   2: core::panicking::assert_failed_inner
   3: core::panicking::assert_failed
             at /usr/src/rustc-1.63.0/library/core/src/panicking.rs:181:5
   4: bkt::cache_tests::stable_hash
             at ./src/lib.rs:869:9
   5: bkt::cache_tests::stable_hash::{{closure}}
             at ./src/lib.rs:868:5
   6: core::ops::function::FnOnce::call_once
             at /usr/src/rustc-1.63.0/library/core/src/ops/function.rs:248:5
   7: core::ops::function::FnOnce::call_once
             at /usr/src/rustc-1.63.0/library/core/src/ops/function.rs:248:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

### PR DESCRIPTION
**Pull request by alexanderkjall**
fixes #39 (I hope, lacking a test machine to verify on)

### CODE REVIEW

**Comment by dimo414** [comment]
Double-checked that the [test is still run on CI](https://github.com/dimo414/bkt/actions/runs/6756454103/job/18365935874?pr=49#step:7:90) so this seems fine to me. Will you be able to confirm this fix after it merges?

**Comment by alexanderkjall** [comment]
Let me track down a machine to test this on before merging, I'll report back.

**Comment by dimo414** [comment]
The change makes sense to me so I don't mind merging first if that's at all easier; I just want to be able to validate the issue is resolved at some point 🙂

File: src/lib.rs [hunk 1]
```diff
@@ -867,6 +867,7 @@ mod cache_tests {
 
     // Sanity-checking that cache_key's behavior is stable over time. This test may need to be
     // updated when changing Rust versions / editions.
+    #[cfg(target_endian = "little")]
```
Comment (dimo414) [inline src/lib.rs:870]:
```suggestion
    // Disabled on hardware that generates other hashes, see #39
    #[cfg(target_endian = "little")]
```

**Comment by dimo414** [comment]
@alexanderkjall still interested in merging this?

**Comment by alexanderkjall** [comment]
yes please, I didn't manage to scare up a test machine and are currently very short on free time :(

**Pull request merged**